test/libyul/yulOptimizerTests/varNameCleaner/reshuffling.yul

```diff
@@ -0,0 +1,12 @@
+{
+	let x_1 := 1
+	let x_2 := 2
+	let x_3 := 3
+}
+// ----
+// varNameCleaner
+// {
+//     let x := 1
+//     let x_2 := 2
```

Comment:
The question is whether we actually *want* ``x_1`` here... 
What does currently happen for
```
{
  let x_4 := 4
  let x_3 := 3
  let x_2 := 2
  let x_1 := 1
}
```
?
Will this become:
```
{
  let x := 4
  let x_1 := 3
  let x_2 := 2
  let x_5 := 1
}
```
?
If so, that's not particularly nice and it may be worth spending the additional effort to avoid those gaps - that should at least be possible by running through the AST twice...

Reply:
The underlying issue is probably why the `Disambiguator` *does* copy the AST...